frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/file/file_block_component.dart

```diff
@@ -406,7 +511,18 @@ class FileBlockComponentState extends State<FileBlockComponent>
     // Remove the file block from the drop state manager
     dropManagerState.remove(FileBlockKeys.type);
 
-    final name = Uri.tryParse(url)?.pathSegments.last ?? url;
+    final uri = Uri.tryParse(url);
+    if (uri == null) {
+      return;
+    }
+
+    String name = uri.pathSegments.isNotEmpty ? uri.pathSegments.last : "";
+    if (name.isEmpty && uri.pathSegments.length > 1) {
+      name = uri.pathSegments[uri.pathSegments.length - 2];
```

Comment:
I'm a little confused about this logic. Why use the last but one element as the name?

Reply:
Consider this url:
`https://appflowy.io/download/` The last element is empty, so we take `download`.

We can change this later to use the meta data for a website.